### PR TITLE
Added support for both new and old API's response structure

### PR DESCRIPTION
**Pull request by ahmedsadman**
The newest version of wit.ai uses a different response structure. In this new structure, "intent" is NOT a separate key, instead it is integrated in "entities". 

So, when new contributors teaches the WIT ai to work with their module, they will face problems because JARVIS uses older api version which uses different response structure.

### CODE REVIEW

File: modules/__init__.py [hunk 1]
```diff
@@ -24,9 +24,10 @@ def process_query(input):
             'Authorization': 'Bearer %s' % WIT_AI_ACCESS_TOKEN
         })
         data = r.json()
-        intent = data['outcomes'][0]['intent']
+        intent = data['outcomes'][0]['intent'] if 'intent' not in data['outcomes'][0]['entities'] else data['outcomes'][0]['entities']['intent'][0]['value']
         entities = data['outcomes'][0]['entities']
-        confidence = data['outcomes'][0]['confidence']
+        confidence = data['outcomes'][0]['confidence'] if 'intent' not in data['outcomes'][0]['entities'] else data['outcomes'][0]['entities']['intent'][0]['confidence']
+        print intent, confidence
```
Comment (horrormyth) [inline modules/__init__.py]: Do we need this?

File: modules/__init__.py [hunk 2]
```diff
@@ -24,9 +24,10 @@ def process_query(input):
             'Authorization': 'Bearer %s' % WIT_AI_ACCESS_TOKEN
         })
         data = r.json()
-        intent = data['outcomes'][0]['intent']
+        intent = data['outcomes'][0]['intent'] if 'intent' not in data['outcomes'][0]['entities'] else data['outcomes'][0]['entities']['intent'][0]['value']
```
Comment (horrormyth) [inline modules/__init__.py]: Could we follow PEP8 please,  https://www.python.org/dev/peps/pep-0008/#id19

Comment (horrormyth) [inline modules/__init__.py]: If we could follow Could we follow PEP8 please, https://www.python.org/dev/peps/pep-0008/#id19 then that would be greate

File: modules/__init__.py (hunk 1 above)
Comment (horrormyth) [inline modules/__init__.py]: Do we need this

**Comment by swapagarwal** [comment]
@ahmedsadman The data-training is done on wit console (independent of the version), and the code that calls Wit API includes the version (in jarvis). I'm not sure I understand how new users will face problems. Can you elaborate?
@horrormyth Thank you for taking the time to review! 🎁 I had included style-check bot a month back, but disabled it temporarily as it seemed to dishearten new contributors.

File: modules/__init__.py (hunk 1 above)
Comment (ahmedsadman) [inline modules/__init__.py]: Sorry, the print statement was for debugging purpose, I removed it in the latest commit.

File: modules/__init__.py (hunk 2 above)
Comment (ahmedsadman) [inline modules/__init__.py]: Commit done

**Comment by astroengisci** [comment]
@swapagarwal I had the same problem when I was writing my dice extension. If you go into wit.ai, create a new bot, and sub in its access key, even though the request includes the version, the format is different. Somehow, "intent" is now included in the "entities" block of the JSON by default, rather than being it's own block right under "outcomes". Just like @ahmedsadman, I had to modify my local code for testing purposes. 

So, your wit.ai returns:

    "outcomes":
    {
        "intent":
        {
            // which module should handle it
        }
        "entities":
        {
            // stuff for the module to handle
        }
    }

However, with a new app made for testing purposes, it returns:

    "outcomes":
    {
        "entities":
        {
            "intent":
            {
                // which module should handle it
            }
            "some_other_entity":
            {
                // stuff for the module to handle
            }
        }
    }

**Comment by swapagarwal** [comment]
@astroengisci Oh, it seems like new apps are getting the new structure. I suspect that since jarvis is an old app, it continues to get the old structure from wit to maintain backwards compatibility. I'll try creating a new app today, and see how it goes. Also, there must be some blog post mentioning these changes. That official link should have complete information on all the api changes in the new version.

**Comment by ahmedsadman** [comment]
@swapagarwal  Thanks for the understanding. Please let me know if there is any update on this so that I can continue working on the soccer module. 

_Note: It's all up to you whether you want to create a new app (which is surely gonna take good amount of effort), but actually it's not necessary, because the code I submitted in this request can handle both new and old structure. In this way, new module creators would not face problem while using the latest version._

**Comment by swapagarwal** [comment]
Opened https://github.com/wit-ai/wit/issues/697

**Pull request closed without merging**